acq4/devices/Pipette/calibration.py

```diff
@@ -59,29 +61,36 @@ def calibratePipette(pipette: Pipette, imager: Camera, scopeDevice, searchSpeed=
         # to see whether we made it back to the desired position (and if not, apply
         # a correction as well as remember the apparent time delay between camera images and
         # position updates)
+        compareFrame = imager.acquireFrames(1).getResult()[0].data().astype(float)
+        comparisonError = ((framesArray.astype(float) - compareFrame[None, ...])**2).sum(axis=1).sum(axis=1)
+        mostSimilarFrame = np.argmin(comparisonError)
+        pipetteCameraDelay = frames[mostSimilarFrame].info()['time'] - frames[centerIndex].info()['time']
+        correctedCenterTime = frames[centerIndex].info()['time'] - pipetteCameraDelay
+        correctedCenterPipPos = getPipettePositionAtTime(posEvents, correctedCenterTime)
+        pipette._moveToGlobal(correctedCenterPipPos, speed='fast').wait()
 
         # record from imager and pipette position while retracting pipette out of frame
         retractPos = centerPipPos - pipVector * 2e-3
         frames2, posEvents2 = watchMovingPipette(pipette, imager, retractPos, speed=searchSpeed, _future=_future)
+        frames2Array = np.stack([f.data() for f in frames2])
```

Comment:
I don't love `frames2` as a variable name, but it's easily refactored later, so don't sweat it.